### DISCUSSION TITLE
ESP-IDF component and Platform I/O library

### DISCUSSION THREAD

**Discussion by Johboh** [nlohmann/json#4250] [General]
Hi!
For those unaware. [ESP-IDF](https://github.com/espressif/esp-idf) is a development framework when working with ESP32 MCUs. [Platform I/O](https://platformio.org/) is an "IDE solution" for working with the ESP-IDF and Arduino framework (and others).

ESP-IDF uses the concept of Components and have a [component registry](https://components.espressif.com/) where one can find shared components to use in a project.
Platform I/O have packages and a [registry](https://registry.platformio.org/) where one can find packages.

I had a wanted to use this json library both as a component for ESP-IDF (not Platform I/O), as well as a Arduino/ESP-IDF library when using Platform I/O. I couldn't find any published official version of this library in any of the mentioned registries, so I made a fork and added workflows for publishing this library as a component as well as a package. This is the only change needed: https://github.com/Johboh/nlohmann-json/commit/f4f239f2fcad50e71c91afd36d75acc8fbb6c86f (and of course tokens for platform I/O and ESP-IDF component registry).

Does components/packages already exists for this library that I missed?
And if not, what do you think of bringing these workflows in?

Regards,
Johan

**Comment by nlohmann** [comment]
I don't want to act as package maintainer, so I would not want to add any workflows to this repository. But if anyone wants to do this, I am happy to add a note to the README.

**Comment by Johboh** [comment]
OK, understandable.
I will continue to publish them from my fork on each new release. I only had to "disable" existing workflows so rebasing should be trivial.

I have to publish them under my namespace, [johboh/nlohmann-json](https://components.espressif.com/components/johboh/nlohmann-json), which is OK, but _nlohmann/json_ would be better. Same for Platform I/O. It's possible for me to publish them under the _nlohmann namespace_ if you want, but for that, you would need to create tokens for me to use in the workflows. I understand if you don't want to or don't have the trust to do so, but that is an option to make it more 'official'.

Regards,
Johan

**Reply by nlohmann** [reply]
I'm fine with publishing it in your namespace.

**Comment by Johboh** [comment]
Ok!
I will continue to do so.

What do you think about adding the component definition to main repo? Eg what is in this folder: https://github.com/Johboh/nlohmann-json/tree/develop/espidf-component
This is similar to the setup for bazel and others, and will enable anyone who download this repo to use and build it for ESP-IDF.

**Reply by nlohmann** [reply]
Is this necessary or can this also be added to the fork?

**Comment by Johboh** [comment]
No it's not needed, it can stay in the fork.
It was more about if you wanted to add support for ESP-IDF components (without publishing) in the main repo.
